lib/_http_server.js

```diff
@@ -449,13 +450,19 @@ function onParserExecute(server, socket, parser, state, ret, d) {
   onParserExecuteCommon(server, socket, parser, state, ret, undefined);
 }
 
+const badRequestResponse = Buffer.from(
+  'HTTP/1.1 400 ' + STATUS_CODES[400] + CRLF + CRLF, 'ascii'
+);
 function socketOnError(e) {
   // Ignore further errors
   this.removeListener('error', socketOnError);
   this.on('error', () => {});
 
-  if (!this.server.emit('clientError', e, this))
+  if (!this.server.emit('clientError', e, this)) {
+    if (this.writable)
+      this.write(badRequestResponse);
```

Comment:
should the destroy be called only after the callback to write is called? Or will that not matter here?

Reply:
Per #6554, it might make sense to split this in 501 responses for `HPE_INVALID_METHOD` errors, 505 for `HPE_INVALID_VERSION`, and 400 responses for everything else:
```js
if (e.code === 'HPE_INVALID_METHOD')
  this.write(notImplementedResponse);
else if (e.code === 'HPE_INVALID_VERSION')
  this.write(httpVersionNotSupportedResponse);
else
  this.write(badRequestResponse);
```

Reply:
I tried it. But in the case of test/parallel/test-http-blank-header.js, HPE_INVALID_METHOD is also returned. I think it's a parser bug.

Reply:
It's a bug in the test that went unnoticed until now.  It sends extra junk at the end of the request that http-parser (rightly) rejects as a malformed request method.

Remove the extraneous `\r\nhello world` and you should be good.